rootex/framework/entity.h

```diff
@@ -53,9 +55,15 @@ class Entity
 	template <class ComponentType = Component>
 	Ref<ComponentType> getComponentFromID(ComponentID ID) const;
 
+	Ref<Script> getScript() { return m_Script; }
+	sol::table getScriptEnvt();
+	void setScriptEnvt(sol::table changed);
+
 	JSON::json getJSON() const;
 	const HashMap<ComponentID, Ref<Component>>& getAllComponents() const;
 	bool isEditorOnly() const { return m_IsEditorOnly; }
+
+	bool call(String function, Vector<Variant> args);
```

Comment:
Take references. Probably will need a `const String&` in the `function` parameter because it needs to bind to constants as well